src/test/java/io/strimzi/kafka/bridge/http/ProducerIT.java

```diff
@@ -1117,4 +1122,98 @@ void sendSimpleMessageWithWrongContentType(VertxTestContext context) throws Inte
+    void dynamicContentTypeHandling(String firstContentType, String secondContentType, VertxTestContext context) throws Exception {
+        final String key = "exampleKey";
+        final String value = "Hello, world!";
+
+        JsonObject record = createDynamicRecord(key, value, firstContentType);
+        JsonArray records = new JsonArray().add(record);
+        JsonObject root = new JsonObject().put("records", records);
+
+        LOGGER.info("Sending:\n{}", root.toString());
+
+        // Send the first request
+        producerService()
+            .sendRecordsRequest(topic, root, firstContentType)
+            .sendJsonObject(root, response -> {
+                if (response.succeeded()) {
+                    HttpResponse<JsonObject> httpResponse = response.result();
+                    context.verify(() -> {
+                        assertThat(httpResponse.statusCode(), is(HttpResponseStatus.OK.code()));
+                        LOGGER.info("Successfully processed record with content type: {}", secondContentType);
+                    });
+                } else {
+                    context.failNow(response.cause());
+                }
+            });
+
+        record = createDynamicRecord(key, value, secondContentType);
+        records = new JsonArray().add(record);
+        root = new JsonObject().put("records", records);
+
+        // Send the second request
+        producerService()
+            .sendRecordsRequest(topic, root, secondContentType)
+            .sendJsonObject(root, response -> {
+                if (response.succeeded()) {
+                    HttpResponse<JsonObject> httpResponse = response.result();
+                    context.verify(() -> {
+                        assertThat(httpResponse.statusCode(), is(HttpResponseStatus.OK.code()));
+                        LOGGER.info("Successfully processed record with content type: {}", secondContentType);
+                        context.completeNow();
+                    });
+                } else {
+                    context.failNow(response.cause());
+                }
+            });
+
+        assertThat(context.awaitCompletion(TEST_TIMEOUT, TimeUnit.SECONDS), is(true));
+    }
+
+    private JsonObject createDynamicRecord(String key, String value, String contentType) {
+        JsonObject record = new JsonObject();
+
+        switch (contentType) {
+            case BridgeContentType.KAFKA_JSON_JSON:
+                // Both key and value are wrapped in a JsonObject
+                record.put("key", new JsonObject().put("actualKey", key));
+                record.put("value", new JsonObject().put("message", value));
+                break;
+            case BridgeContentType.KAFKA_JSON_BINARY:
+                // Both key and value are encoded in Base64
+                String base64Key = Base64.getEncoder().encodeToString(key.getBytes());
+                String base64Value = Base64.getEncoder().encodeToString(value.getBytes());
```

Comment:
do we need these two additional local vars?

Reply:
I think not will remove it